src/payment/data/service.js

```diff
@@ -59,23 +60,10 @@ function getOrderType(productType) {
 }
 
 export function transformResults(data) {
-  const results = camelCaseObject(data) || {};
+  const results = camelCaseObject(data);
```

Comment:
If data is `undefined` or `null`, I think it's better for us to blow up here than it is for us to blow up somewhere far away downstream when transformResults gives the app data that's sparse.

Ideally we might log something here and bail in a nice way, but this should now at least cause us to show a fallback-error.  Before it would just pass a really weird sparse "basket" back to the app.